### PR TITLE
Ltac2.Ltac1: add to/of_preterm

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot run full ci

File: user-contrib/Ltac2/Ltac1.v [hunk 1]
```diff
@@ -40,6 +40,10 @@ Ltac2 @ external apply : t -> t list -> (t -> unit) -> unit := "coq-core.plugins
 Ltac2 @ external of_constr : constr -> t := "coq-core.plugins.ltac2_ltac1" "ltac1_of_constr".
 Ltac2 @ external to_constr : t -> constr option := "coq-core.plugins.ltac2_ltac1" "ltac1_to_constr".
 
+(** [preterm] is called [uconstr] in Ltac1. *)
+Ltac2 @ external of_preterm : constr -> t := "coq-core.plugins.ltac2_ltac1" "ltac1_of_preterm".
+Ltac2 @ external to_preterm : t -> constr option := "coq-core.plugins.ltac2_ltac1" "ltac1_to_preterm".
```
[the remark this answers is not in the document]
Comment (JasonGross) [reply user-contrib/Ltac2/Ltac1.v:45]: (Also, is there no way to get typechecking on these?)

[the remark this answers is not in the document]
Comment (ppedrot) [reply user-contrib/Ltac2/Ltac1.v:45]: Good catch. As for your question there is `Constr.pretype`.

[the remark this answers is not in the document]
Comment (JasonGross) [reply user-contrib/Ltac2/Ltac1.v:45]: No, I mean, no way to get `coqc` to complain about incorrect types on `Ltac2 @ external` declarations

[the remark this answers is not in the document]
Comment (ppedrot) [reply user-contrib/Ltac2/Ltac1.v]: I thought there was a lightweight check but apparently I was wrong.

[the remark this answers is not in the document]
Comment (JasonGross) [reply user-contrib/Ltac2/Ltac1.v]: Created #18553 to track this

**Comment by ppedrot** [comment]
@coqbot run full ci

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**